lib/src/utils.dart

```diff
@@ -43,3 +43,53 @@ YamlWarningCallback yamlWarningCallback = (message, [SourceSpan? span]) {
   if (span != null) message = span.message(message);
   print(message);
 };
+
+// The following utility functions are copied from the string_scanner package.
```

Comment:
Only copy what's needed. Some of these aren't used.

I think the following should be sufficient:
```dart
// The following utility functions are copied from the string_scanner package,
// and modified slightly.
//
// See https://en.wikipedia.org/wiki/UTF-16#Code_points_from_U+010000_to_U+10FFFF
// for documentation on how UTF-16 encoding works and definitions of various
// related terms.

/// The start of the UTF-16 high surrogate block.
const _highSurrogateLowerBound = 0xD800;

/// The start of the UTF-16 low surrogate block.
const _lowSurrogateLowerBound = 0xDC00;

/// Mask for the encoded bits of a surrogate value.
const _surrogateValueMask = 0x3FF;

/// Whether [codeUnit] is a UTF-16 high surrogate.
bool isHighSurrogate(int codeUnit) =>
    codeUnit ^ _highSurrogateLowerBound <= _surrogateValueMask;

/// Whether [codeUnit] is a UTF-16 low surrogate.
bool isLowSurrogate(int codeUnit) =>
    codeUnit ^ _lowSurrogateLowerBound <= _surrogateValueMask;
```

Copying something that we don't need just means more code to maintain, which we won't (I see no tests), and someone might one day start assuming that it works. Just keep it simple.